easybuild/tools/filetools.py

```diff
@@ -1162,6 +1163,50 @@ def rmtree2(path, n=3):
     else:
         _log.info("Path %s successfully removed." % path)
 
+def find_backup_name_candidate(src_file):
+    """Returns a non-existing file to be used as destination for rotating backups"""
+    dst_file = src_file
+
+    if os.path.exists(src_file):
+        i = 0
+        dst_file = "%s_%d" % (src_file, i)
+        while os.path.exists(dst_file):
+            i += 1
+            dst_file = "%s_%d" % (src_file, i)
+
+    return dst_file
+
+
+def back_up_file(src_file, backup_extension="", hidden=False):
+    """
+    Backs up a file appending a backup extension and a number to it (if there is already an existing backup). Returns
+    the name of the backup
+
+    :param src_file: file to be back up
+    :param backup_extension: optional extension of the backup
+    :param hidden: make a hidden (leading dot) backup
+    """
+    if hidden:
+        leading_string = "."
+    else:
+        leading_string = ""
+
+    if backup_extension:
+        trailing_string = ".%s" % backup_extension
+    else:
+        trailing_string = ""
```

Comment:
Shorter/cleaner (imho):

```python
fn_pref, fn_suff = '', ''
if hidden:
    fn_pref = '.'
if backup_extension:
    fn_suff = '.%s' % backup_extension
```

Also, I prefer using `''` rather than `""` for strings without spaces (I find it easier on the eyes), but that's nitpicking ;)